modules/account-tokenizer/src/lib.rs

```diff
@@ -0,0 +1,417 @@
+			T::Currency::reserve_named(&RESERVE_ID, &who, T::MintRequestDeposit::get())?;
+
+			// Submit confiramtion call to be serviced by foreign state oracle
+			let call: <T as Config>::Call = Call::<T>::confirm_mint_request {
+				owner: who.clone(),
+				account: account.clone(),
+			}
+			.into();
+			T::ForeignStateQuery::query_task(who.clone(), call.using_encoded(|x| x.len()), call)?;
+
+			Self::deposit_event(Event::MintRequested { account, who });
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn confirm_mint_request(
+			origin: OriginFor<T>,
+			owner: T::AccountId,
+			account: T::AccountId,
+		) -> DispatchResult {
+			// Extract confirmation info from Origin.
+			let data = T::OracleOrigin::ensure_origin(origin)?;
+			let rejected = data.get(0).ok_or(Error::<T>::BadOracleData)?.is_zero();
+
+			// Accept or reject the mint request.
+			if rejected {
+				Self::reject_mint_request(owner)
+			} else {
+				Self::accept_mint_request(owner, account)
+			}
+		}
+
+		// Requests to burn an Account Token. Sends XCM message to the relaychain to transfer the
+		// control of the account.
+		// No storage change is done, until confirmed by the Oracle.
+		#[pallet::weight(0)]
+		pub fn request_burn(origin: OriginFor<T>, account: T::AccountId, new_owner: T::AccountId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let nft_class_id = Self::nft_class_id();
+
+			// Obtain info about the account token.
+			let token_id = Self::minted_account(&account).ok_or(Error::<T>::AccountTokenNotFound)?;
+			let owner = T::NFTInterface::owner(&nft_class_id, &token_id).ok_or(Error::<T>::AccountTokenNotFound)?;
+
+			// Ensure that only the owner of the NFT can burn.
+			ensure!(who == owner, Error::<T>::CallerUnauthorized);
+
+			// The XCM fee is burned.
+			T::Currency::withdraw(
+				&who,
+				T::XcmInterface::get_transfer_proxy_xcm_fee(),
+				WithdrawReasons::FEE,
+				KeepAlive,
+			)?;
+
+			// Send an XCM to relaychain to relinquish the control of the `account` to `new_owner`.
+			T::XcmInterface::transfer_proxy(account.clone(), new_owner.clone())?;
+
+			// Submit confiramtion call to be serviced by foreign state oracle
+			let call: <T as Config>::Call = Call::<T>::confirm_burn_account_token {
+				account: account.clone(),
+				new_owner: new_owner.clone(),
+			}
+			.into();
+			T::ForeignStateQuery::query_task(who.clone(), call.using_encoded(|x| x.len()), call)?;
+
+			Self::deposit_event(Event::BurnRequested {
+				account,
+				owner,
+				token_id,
+				new_owner,
+			});
+			Ok(())
+		}
+
+		/// Confirm the parachain account has relinquished the control of the account on the
+		/// relaychain to the `new_owner`. The NFT can now be burned.
+		///
+		/// Only callable by the Oracle.
+		#[pallet::weight(0)]
+		#[transactional]
+		pub fn confirm_burn_account_token(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			new_owner: T::AccountId,
+		) -> DispatchResult {
+			T::OracleOrigin::ensure_origin(origin)?;
```

Comment:
this could fail if the requester is making duplicated requests, or we should ensure duplicated requests are not possible

Reply:
The NFT token can only be burned once - repeated burn will fail the transaction. I think we are safe